server/task.go

```diff
@@ -604,6 +604,17 @@ func (s *Server) changeTaskStatusWithPatch(ctx context.Context, task *api.Task,
 				zap.Error(err),
 			)
 		}
+		// Sync task.database_id with the created database.ID
+		taskDatabaseIDPatch := &api.TaskPatch{
```

Comment:
I am a bit baffled. So for creating database task, it's expected to not have database ID.

Add to set database ID after task completes is also pending discussion as it introduces a new pattern that we will mutate the task domain specific field after task state changes.

What makes me further wonder is adding the database id here would make the new database surface in the database list sidebar you suggest. Since they are not quite related, we are changing a task state, how could it affect the global database list?

So I suspect the story here is because we patch the database id here, and during the next polling from the issue detail page (search pollOnCreateStateChange), it fetches the task, which in turns fetches the database, and thus fills the database list stored in the frontend.

So there are probably 2 ways to fix this:

1. The one we used here, but we really need to provide the detailed comment about the story.
2. We fix this in IssueSidebar, we hook the place where we notice the creation database task completes, and refresh the database list (we don't have a fetchDatabaseByName method, but we can use fetchDatabaseListByInstanceId, though a bit inefficient).

I am fine with using 1 since that's simpler to implement, however, it's not intuitive as option 2 so we need to add very detailed comment.

Reply:
Before I look up the backend, I almost want to fix it with fetchAllDatabase just in frontend. But after I had some talks with Danny, I think the `task` should have its related `database_id` both for recording created and patched database.

Reply:
This is a fine approach, just we need to add detailed comment.

Also don't forget to update the database_id comment, both in the schema file and the task definition, it currently says it's empty for creating database.

Reply:
Also include some offline discussion here: if we take this approach, we may want to move sqlite database row creation to task_execute_database_create.go, otherwise there may still be some race condition in user story.
broken story ordering: 1) we change task to done, 2) frontend thinks pipeline/task is done and fetches the database, and it's still null, 3) backend creates the database row and updates the issue task database_id, but it's too late.

We could eliminate the race condition by moving database row creation to task_execute_database_create.go. Also currently, we fail-open for database row creation failure and letting schema syner to set it. Now we have to fail-close on database row creation failure, but we need to figure out any possible cases that could cause failure, e.g. row already-exist, etc.